File: app/src/org/musicbrainz/mobile/view/HtmlAssetTextView.java
package org.musicbrainz.mobile.view;

import org.musicbrainz.mobile.util.Utils;

import android.content.Context;
import android.text.Html;
import android.util.AttributeSet;
import android.widget.TextView;

public class HtmlAssetTextView extends TextView {
    
    private Context context;
    
    public HtmlAssetTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
    }
    
    public void setAsset(String asset) {
        setText(Html.fromHtml(Utils.stringFromAsset(context, asset)));
    }

}

File: app/src/main/java/org/metabrainz/android/presentation/view/HtmlAssetTextView.kt
package org.metabrainz.android.presentation.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatTextView
import androidx.core.text.HtmlCompat
import org.metabrainz.android.util.Utils

class HtmlAssetTextView(context: Context, attrs: AttributeSet?) : AppCompatTextView(context, attrs) {
    fun setAsset(asset: String?) {
        text = HtmlCompat.fromHtml(Utils.stringFromAsset(context, asset), HtmlCompat.FROM_HTML_MODE_LEGACY)
    }
}